fletcher/base.py

```diff
@@ -741,6 +735,17 @@ def take(self, indices, allow_fill=False, fill_value=None):
         result = take(data, indices, fill_value=fill_value, allow_fill=allow_fill)
         return self._from_sequence(result, dtype=self.data.type)
 
+    def unique(self):
+        """
+        Compute the ExtensionArray of unique values.
+        It completely relies on the Pyarrow.ChunkedArray.unique
+
+        Returns
+        -------
+        uniques : ExtensionArray
+        """
+        return type(self)(self.data.unique())
```

Comment:
`unique` is not implemented for all arrow types. So maybe could put a try except around this and still fall back to the naive implementation otherwise?

Reply:
Do you know which type is not implemented ? 
Yes I'll do that, thanks !

Reply:
I think typically nested types like list and struct type.

Reply:
yes, you are right :
```python
In [3]: list_pa_array = pa.array([[1, 1], [], [1, 1], [1, 2], []])                                                                          

In [4]: list_pa_array.unique()                                                                                                              
---------------------------------------------------------------------------
ArrowNotImplementedError                  Traceback (most recent call last)
<ipython-input-4-7b7dd44ca4e0> in <module>
----> 1 list_pa_array.unique()

~/Projects/miniconda3/envs/py38/lib/python3.8/site-packages/pyarrow/array.pxi in pyarrow.lib.Array.unique()

~/Projects/miniconda3/envs/py38/lib/python3.8/site-packages/pyarrow/error.pxi in pyarrow.lib.check_status()

ArrowNotImplementedError: unique not implemented for list<item: int64>
```
but it's not going to work with pandas neither 
```python
In [9]: pd.unique(list_pa_array.to_pandas())                                                                                                
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-9-1f88df243e1f> in <module>
----> 1 pd.unique(list_pa_array.to_pandas())

~/Projects/miniconda3/envs/py38/lib/python3.8/site-packages/pandas/core/algorithms.py in unique(values)
    403 
    404     table = htable(len(values))
--> 405     uniques = table.unique(values)
    406     uniques = _reconstruct_data(uniques, dtype, original)
    407     return uniques

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.unique()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable._unique()

TypeError: unhashable type: 'numpy.ndarray'
```
Is there another example of dtype that would work with pandas but not with pyarrow ?

Reply:
Ah yes. And for struct type it will also fail in pandas (dicts are also not hashable)